build-tools/packages/build-cli/src/handlers/stateHandlers.ts

```diff
@@ -9,21 +9,52 @@ import { CommandLogger } from "../logging.js";
 import { MachineState } from "../machines/index.js";
 
 /**
- * An async function that handles state. Typically this type is used via the {@link BaseStateHandler} class and its
- * subclasses.
+ * An async function that is called to handle a particular state from a jssm state machine. Typically this type is used
+ * via the {@link BaseStateHandler} class and its subclasses.
+ *
+ * The function receives a current state, the state machine itself, a testMode boolean, and a `data` object of any type
```

Comment:
Nit: I think you can use `@param` comments for function types, FYI. Might make the docs a bit easier to parse.

Love the extra details here!